hls4ml/batch/do_hls_things.py

```diff
@@ -170,22 +173,39 @@ def sigmoid(x):
 
 proj_loc = out_loc_name + f'/{hls_model_name}/myproject_prj/'
 
-# fpgapart = 'xcu250-figd2104-2L-e'
-fpgapart = 'xcvu13p-fsga2577-2-e'
-
 if 'tcn' in mod_name or 'TCN' in mod_name:
     print('~~~ TCN is in the name, so will assume it is a CNN ~~~')
-    cfg = hls4ml.converters.create_config(backend='Vivado')
-    cfg['IOType']     = 'io_stream' # Must set this if using CNNs!
+    cfg = hls4ml.converters.create_config(backend='VivadoAccelerator')
+    cfg['IOType']     = 'io_stream' # Must set this if using CNNs!                                                                                                               
     cfg['HLSConfig']  = config
     cfg['KerasModel'] = model
     cfg['OutputDir']  = proj_loc
-    cfg['XilinxPart'] = fpgapart
+    cfg['ClockPeriod']= 5
+    cfg['XilinxPart'] = 'xczu9eg-ffvb1156-2-e'
+    cfg['Board']      = 'zcu102'
+    cfg['ProjectName'] = 'myproject'
+    cfg['InputData']  = args.data.split(',')[0]
+    if os.path.exists(args.data.split(',')[1].replace('.npy','_hls.npy')):
+        print ("using HLS predicted value")
+        print("-----------------------------------")
+        cfg['OutputPredictions'] = args.data.split(',')[1].replace('.npy','_hls.npy')
+    else :
+        print ("using Keras predicted values, run again to load HLS predicted values")
+        print("-----------------------------------")
+        cfg['OutputPredictions'] = args.data.split(',')[1]
+    print(cfg)
     hls_model = hls4ml.converters.keras_to_hls(cfg)
 else:
-    hls_model = hls4ml.converters.convert_from_keras_model(model, part=fpgapart,
+    print("-----------------------------------")
+    print("CAREFUL!!! this code has been optimized for TCN only")
+    hls_model = hls4ml.converters.convert_from_keras_model(model, part='xczu9eg-ffvb1156-2-e',
                                                        hls_config=config,
-                                                       output_dir=proj_loc)
+                                                       backend='VivadoAccelerator',
```

Comment:
Define backend before and use it when doing create_config

Reply:
ok